db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ColumnDefinitionNode.java

```diff
@@ -669,6 +660,15 @@ void validateDefault(DataDictionary dd, TableDescriptor td)
 							  defaultNode.getDefaultText(), 
 							  defaultValue);
 
+			// for fixed-length char type, if default value cannot fix in the column, for example,
+			// default of 'xxxxx' cannot fix in the type of CHAR(4), for such cases, we should error out.
+			// If default value after trimming the trailing space can fit in CHAR(4), it can still go
+			// through
+			if (columnTypeId.getTypeFormatId() == StoredFormatIds.CHAR_TYPE_ID && defaultValue != null) {
+				DataValueDescriptor newDefault = defaultValue.cloneValue(false);
+				newDefault.normalize(getType(), newDefault);
+			}
+
```

Comment:
Why a clone is needed for normalization

Reply:
The code here is a test to see if the default value actually is compliant to the column definition or not. I wanted to keep the user input default value intact from the test, thus made a clone. Thanks.